app/scripts/ui.js

```diff
@@ -11,19 +11,62 @@ import Eth from 'ethjs';
 import EthQuery from 'eth-query';
 import StreamProvider from 'web3-stream-provider';
 import log from 'loglevel';
+import fetch from 'node-fetch';
 import launchMetaMaskUi from '../../ui';
 import {
   ENVIRONMENT_TYPE_FULLSCREEN,
   ENVIRONMENT_TYPE_POPUP,
 } from '../../shared/constants/app';
+import { SUPPORT_LINK } from '../../ui/helpers/constants/common';
 import ExtensionPlatform from './platforms/extension';
 import { setupMultiplex } from './lib/stream-utils';
 import { getEnvironmentType } from './lib/util';
 import metaRPCClientFactory from './lib/metaRPCClientFactory';
+import getFirstPreferredLangCode from './lib/get-first-preferred-lang-code';
 
 start().catch(log.error);
 
 async function start() {
+  const preferredLocale = await getFirstPreferredLangCode();
+
+  const url = browser.runtime.getURL(
+    `../_locales/${preferredLocale}/messages.json`,
+  );
+  const messages = await (await fetch(url)).json();
+
+  function displayCriticalError(container, err, msg) {
+    const html = `
+    <style>
+      #app-content {margin: 10px;}
+      .critical-error {
+        color:red; 
+        background-color:yellow; 
+        text-align: left
+      }
+      blockquote {
+        margin: 20px 0 20px 5px; 
+        padding-left: 5px; 
+        border-left: 1px solid grey
+      }  
+      a {color: blue}
+      a:hover {color: green}
+    </style>
+    <div class="critical-error">
+      ${messages[msg].message}
```

Comment:
@Gudahtt We could use your advice here. We are displaying an error message to the user by directly setter `innerHTML` (see below). This is happening in a place where we don't have access to react. A new message has been added.

Recommendations for how to get this to pass the locale verification script?

Reply:
The verify locales script look for `t('[key]')`. It doesn't know what `t` is, it just looks for a function call with that name. We can create an equivalent function here.

See here for the definition of `t`: https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/i18n.js#L18

Reply:
Also it's critical that we have a fallback for if the localization fails, or if the attempt to download the localized messages fails, or anything like that. All errors need to be caught, otherwise we'll still get the blank screen. Showing the wrong language is better than showing a blank screen.